src/runtime-rs/crates/resource/src/share_fs/sandbox_bind_mounts.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2023 Alibaba Cloud
+// Copyright (c) 2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Note:
+// sandbox_bind_mounts supports kinds of mount patterns, for example:
+// (1) "/path/to", with default readonly mode.
+// (2) "/path/to:ro", same as (1).
+// (3) "/path/to:rw", with readwrite mode.
+//
+// sandbox_bind_mounts: ["/path/to", "/path/to:rw", "/mnt/to:ro"]
+//
+
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use anyhow::{anyhow, Context, Result};
+
+use super::utils::{do_get_host_path, mkdir_with_permissions};
+use kata_sys_util::{fs::get_base_name, mount};
+use kata_types::config::TomlConfig;
+
+const SANDBOX_BIND_MOUNTS_DIR: &str = "sandbox-mounts";
+const SANDBOX_BIND_MOUNTS_RO: &str = ":ro";
+const SANDBOX_BIND_MOUNTS_RW: &str = ":rw";
+
+#[derive(Clone, Default, Debug)]
+pub struct SandboxBindMounts {
```

Comment:
I think maybe it's unnecessary to create a struct to store `sid` and `toml`, you can pass them as parameters.